cloudsync/event.py

```diff
@@ -31,49 +30,72 @@ class Event:
     new_cursor: Optional[str] = None
 
 
+class ProviderGuard(set):
+    def add(self, provider):
+        if provider in self:
```

Comment:
we should consider adding a `clone()` function to providers.

Reply:
that definitely crossed my mind as I was working on this piece -- decided on the simpler solution for now, will make `clone()` a separate task